frontend/src/pages/comparisons/Comparison.tsx

```diff
@@ -15,10 +29,36 @@ const ComparisonPage = () => {
   const searchParams = new URLSearchParams(location.search);
   const series: string = searchParams.get('series') || 'false';
 
-  const { options } = useCurrentPoll();
+  const { options, name: pollName } = useCurrentPoll();
   const tutorialLength = options?.tutorialLength ?? 0;
   const tutorialAlternatives = options?.tutorialAlternatives ?? undefined;
   const tutorialDialogs = options?.tutorialDialogs ?? undefined;
+  const [alreadyMadeComparisons, setAlreadyMadeComparisons] = useState<
+    Array<string>
+  >([]);
+
+  let dialogs;
+  if (tutorialDialogs) {
+    dialogs = tutorialDialogs(t);
+  }
+
+  /**
+   * If 'series' is present in the URL parameters, retrieve the user's
+   * comparisons to avoid suggesting couple of entities that have been already
+   * compared.
+   */
+  useEffect(() => {
+    async function getUserComparisonsAsync(pName: string) {
+      const comparisons = await getUserComparisons(pName);
+      setAlreadyMadeComparisons(
+        comparisons.map((c) => c.entity_a.uid + '/' + c.entity_b.uid)
+      );
+    }
+
+    if (series === 'true') {
+      getUserComparisonsAsync(pollName);
+    }
+  }, [pollName, series]);
```

Comment:
As it's used in a series only, would not it make sense to move this effect into `ComparisonSeries`?

Reply:
At first I thought it was a good idea to receive the existing comparisons as a prop, so that the parent component can control which comparisons will be avoided by the series, but I don't see which use cases would require such a system.

I'm going to move this effect in `<ComparisonSeries>`.

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/705/commits/aa5edac2e780ebbb16249a46fce15689afdccf6b.

The code is a bit more complex now, mainly because I needed to call `genInitialComparisonParams` after `getAlternativesAsync` and `getUserComparisonsAsync`. I chained three promises in the `useEffect`, and it works, but I don't know if this is elegant or the React way to do it.